site/src/pages/release-notes/1.19.0.mdx

```diff
@@ -0,0 +1,190 @@
+    <appender name="RCEA" class="com.linecorp.armeria.common.logback.RequestContextExportingAppender">
+      ...
+      <export>req.cause</export>  <!-- 👈👈👈 -->
+      <export>res.cause</export>  <!-- 👈👈👈 -->
+      ...
+    </appender>
+  </configuration>
+  ```
+- You can now trace a <type://RequestContext> leak by specifying `-Dcom.linecorp.armeria.requestContextLeakDetectionSampler=<sampler-spec>`
+  or implementing a custom <type://FlagsProvider>. #4100 #4232
+  ```java
+  public final class LeakDetectingFlagsProvider implements FlagsProvider {
+    @Override
+    public Sampler<? super RequestContext> requestContextLeakDetectionSampler() {
+      // Samples all request contexts.
+      return Sampler.always();
+    }
+  }
+  ```
+- <type://DocService> now automatically generates a specification of an annotated service for well known types. #4309 #4322
+  - Java POJO
+  - Kotlin data class
+  - Scala case class
+  - Protocol Buffers' `Message`
+  - ScalaPB's `GeneratedMessage`
+- You can now upload files to your <type://GraphqlService> when using 
+  [GraphQL multipart requests](https://github.com/jaydenseric/graphql-multipart-request-spec). #3971 #4199
+- You can now use `armeria-version-catalog` module to easily import Armeria dependencies
+  using [Gradle's version catalog](https://docs.gradle.org/current/userguide/platforms.html#sec:version-catalog-plugin). #4377
+  ```groovy
+  // in settings.gradle
+  dependencyResolutionManagement {
+    versionCatalogs {
+      create("armeriaDeps") {
+        from("com.linecorp.armeria:armeria-version-catalog:1.19.0")
+      }
+    }
+  }
+  ```
+
+## 📈 Improvements
+
+- `CompositeException` now shows full stack traces for all nested exceptions 
+  when `-Dcom.linecorp.armeria.verboseExceptions=true` is set. #4347 #4356
+
+## 🛠️ Bug fixes
+
+- You can now use Armeria Spring actuator with WebFlux correctly. #4396 #4408
+- You no longer see `ClassCastException` when an invalid path request is received. #4405
+- Annotated services now correctly return `204 No Content` status when `void`, `Publisher<Void>` or
+  `CompletionStage<Void>` and `@ProducesJson` are used at the same time. #4390 #4398
+  ```java
+  // 200 OK status and "null" string body is returned
+  @Get("/null-content")
+  @ProducesJson
+  public Publisher<Void> nullBody() {
+    return Mono.empty();
+  }
+  
+  // 204 No Content status and no body is returned
+  @Get("/no-content")
+  @StatusCode(204)
+  @ProducesJson
+  public Publisher<Void> noBody() {
+    return Mono.empty();
+  }
+  ```
+- Armeria server no longer sends chunked encoding header with `304 Not Modified` status. #4386
+- HTTP/2 graceful shutdown timeout is correctly set using <type://ClientFactory#idleTimeoutMillis()>. #4381
+- Unframed <typeplural://GrpcService> now support `application/x-protobuf` and `application/x-google-protobuf`
+  media types. #4355 #4364
+- You can now use a custom `JsonFormat.Printer` for <type://ProtobufResponseConverterFunction>  #4301 #4336
```

Comment:
```suggestion
- You can now use a custom `JsonFormat.Printer` for <type://ProtobufResponseConverterFunction>.  #4301 #4336
```

Reply:
I was confused about this. 😅
<type://...?full> does not work if `.` is next to it.